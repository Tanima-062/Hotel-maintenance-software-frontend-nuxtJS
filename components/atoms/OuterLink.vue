<template>
  <a
    class="text-black-light cursor-pointer underline focus:shadow-outline focus:outline-none"
    rel="noopener"
    :href="href"
    :target="target"
    ><slot
  /></a>
</template>
<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator'

@Component
export default class extends Vue {
  @Prop({ default: '' })
  href!: string

  @Prop({ default: '' })
  target!: string
}
</script>
