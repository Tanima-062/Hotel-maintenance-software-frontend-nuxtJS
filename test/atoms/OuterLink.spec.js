import { shallowMount } from '@vue/test-utils'
import OuterLink from '@/components/atoms/OuterLink.vue'

describe('Atoms OuterLink Check', () => {
  test('OuterLink Test', () => {
    const wrapper = shallowMount(OuterLink)

    // check if it is a vue element
    expect(wrapper.vm).toBeTruthy()

    // check Snapshot
    expect(wrapper.element).toMatchSnapshot()
  })
})
