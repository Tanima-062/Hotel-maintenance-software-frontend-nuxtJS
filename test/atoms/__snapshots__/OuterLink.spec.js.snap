// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms OuterLink Check OuterLink Test 1`] = `
<a
  class="text-black-light cursor-pointer underline focus:shadow-outline focus:outline-none"
  href=""
  rel="noopener"
  target=""
/>
`;
